### ISSUE TITLE
Routing to GPX track. Starting mid track.

### ISSUE THREAD

**Issue by Sw33tMon3yD34l** [osmandapp/osmand-api-demo#18]
It seems that if I try using turn by turn routing to follow an imported GPX track when I launch the routing it wants me to start at the beginning of the track only.

**Use case problem:** My friend creates a GPX for a route from his house to a destination. I want to follow that route to the destination from my house but my house is already half way to the destination and I don't want to go back to his house to start the route. I would like to start in the middle of the route and follow to the end. Current state seems to only allow me to follow the route if I go back to the start and begin from there.

**Proposal:** When I start the routing instead of sending me to the first waypoint I drive until my current location matches one of the waypoints on the track and it proceeds to route me from that way point to the end of the track bypassing the section I am trying to skip.

Additional Benefit: If during routing I decide the current route will take too long and I decide to take a short cut to eliminate a portion of the route this method of resuming will allow me to rejoin the track at any point and still continue to the end.

Basically the current system feels very rigid and not very user friendly. This modification will eliminate a lot of finicky hassle with route navigation.

if someone is willing to take this on I would be more than glad to host a phone call to discuss this and help clarify anything that may seem unclear with the approach. Please message me for my phone number if required.

Thanks in advance!

**Comment by Sw33tMon3yD34l** [comment]
PS. I am not a (very good) developer. I am a product designer and as such I do more process related design than coding. I think I could really help improve the current state if a good developer was willing to work with me on this.

**Comment by vshcherb** [comment]
I think there should be a parameter and it shouldn't start from beginning. If you start navigation then calculation will be from current location, but if you calculate the route then it will include original start point.

I think API should work exactly as OsmAnd standalone works today, in case we want to change it we need to find a proper parameter I guess.

**Comment by Sw33tMon3yD34l** [comment]
I am not sure I understand your comment "If you start navigation then calculation will be from current location, but if you calculate the route then it will include original start point."

I am not using OsmAnd to create routing for me. I am using an imported GPX track and routing to that track. Thus it wants to use the start point of the track only. The more intelligent design would be to allow me to continue from my current location. If my current location is not on the track it should wait until I intersect the track at any point and continue from there. Once I intersect the track it would allow me to navigate to the beginning or end along the track.

**Comment by Sw33tMon3yD34l** [comment]
Here is my example. My friend planned a route in Google maps from Frankfurt (point A) to Zurich (point B). We are riding to Zurich to meet friends. I live in Stuttgart. I am going to ride from Stuttgart to meet him half way along his route. On the day of the ride he gets sick and cannot ride so I want to ride to Zurich using the route he has planned anyway. I would like to be able to ride west and once I intersect the route I would like to be able to continue that route to Zurich. Currently the app (OsmAnd for iOS) forces me to ride all the way back to Frankfurt to navigate the route. This is several hundred Km out of my way. I imagine a better way to have the app process this task would be to:

Capture my location
Loop through all points in track to see if my location is on track.
If not, wait until my location updates and then compare with points on track to see if I am on track now.
If I am on track continue navigation from current pojnt to end of track.

![Screen Shot 2019-07-22 at 6 04 15 PM](https://user-images.githubusercontent.com/20591127/61668909-52804080-acac-11e9-9a49-84181feb6ad2.png)

**Comment by Sw33tMon3yD34l** [comment]
@vshcherb - Thank you. Your comment put me on the right path. To follow an imported track and start from the middle of the track I needed to go to the routing settings and turn off the setting called "Pass along entire track". This was forcing me to use all the track points which is not what I wanted.

Perhaps it makes sense to have this setting turned off by default so newer users arent tripped up by this setting. I imagine there would be many people who might turn off the app mid routing (to stop for lunch or whatever) and when they try and start it again be told to go back to the start. A new user might not find this setting easy to understand or use.

Thanks again for your guidance and time.

**Comment by vshcherb** [comment]
Ok I see cause it is issue is not related for api so I close it and put internal not for this case
